staging/src/k8s.io/apiextensions-apiserver/pkg/features/kube_features.go

```diff
@@ -79,5 +81,5 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	CustomResourceSubresources:      {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
 	CustomResourceWebhookConversion: {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
 	CustomResourcePublishOpenAPI:    {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
-	CustomResourceDefaulting:        {Default: true, PreRelease: featuregate.Beta},
+	CustomResourceDefaulting:        {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
```

Comment:
drop check for the feature gate in validation.go#allowDefaults

Reply:
done

Reply:
note that I only removed the gate check. The `allowDefaults` logic is still needed as it drives the default validation for non-structural, non-pruning CRDs.